### ISSUE TITLE
change cast syntax to go/c++/python style

### ISSUE THREAD

**Issue by mccanne** [brimdata/zq#1707]
Now that we no longer use the zeek, pascal-style type syntax, we shall change the type-cast syntax to the  go/c++/python style, e..g, `type(foo)`.  This will simplify the syntax for primitive types and slightly complicate the runtime since you can't tell if something is a legal cast using a type-alias until runtime (same is true for `:type` style cast but it happens in a different place).

**Comment by mccanne** [comment]
This was implemented in #2427

**Comment by philrz** [comment]
Verified in `zq` commit 40bd1c4.

The prior cast syntax (such as in GA `zq` tagged `v0.29.0`) used `:` followed by the type name.

```
$ echo '{"my_ip": "192.168.0.1"}' | zq -f zson  'put actual_ip=my_ip:ip' -
{
    my_ip: "192.168.0.1",
    actual_ip: 192.168.0.1
} (=0)
```

Now with `zq` commit 40bd1c4, the style uses the type name followed by `()`.

```
$ echo '{"my_ip": "192.168.0.1"}' | zq -f zson  'put actual_ip=ip(my_ip)' -
{
    my_ip: "192.168.0.1",
    actual_ip: 192.168.0.1
}
```

Thanks @mccanne!
